sdk/appcenter/src/test/java/com/microsoft/appcenter/AppCenterTest.java

```diff
@@ -850,6 +849,28 @@ public void setLogUrl() {
         verify(mChannel).setLogUrl(logUrl);
     }
 
+    @Test
+    public void setLogUrlWithNoAppSecret() throws Exception {
+        OneCollectorChannelListener listener = mock(OneCollectorChannelListener.class);
+        whenNew(OneCollectorChannelListener.class).withAnyArguments().thenReturn(listener);
+
+        /* Change log URL before start. */
+        String logUrl = "http://mock";
+        AppCenter.setLogUrl(logUrl);
+
+        /* No effect for now. */
+        verify(listener, never()).setLogUrl(logUrl);
+
+        /* Start should propagate the log URL without App Secret. */
+        AppCenter.start(mApplication, DUMMY_TARGET_TOKEN_STRING, DummyService.class);
```

Comment:
Maybe make 2 variants of this test (sharing most of the code) as this works also with no secret at all.